datasetinsights/datasets/coco.py

```diff
@@ -143,6 +173,87 @@ def _get_local_annotations_zip(self):
     def _get_local_images_zip(self):
         return os.path.join(self.root, f"{self.split}2017.zip")
 
+    def _get_label_mappings(self):
+        """get label mappings.
+
+        Returns:
+            dict: A dict containing {label_id: label_name} mappings.
+        """
+        ann_file_name = (
+            Path(self.root) / "annotations" / f"instances_{self.split}2017.json"
+        )
+        label_mappings = {}
+        with open(ann_file_name, "r") as ann_file:
+            anns = json.load(ann_file)
+            for cat in anns["categories"]:
+                label_mappings[cat["id"]] = cat["name"]
+        return label_mappings
+
+    @staticmethod
+    def _preprocess_dataset(data_path, split):
+        """ Preprocess dataset inside data_path and un-archive if necessary.
+
+            Args:
+                data_path (str): Path where dataset is stored.
+                split (str): indicate split type of the dataset (train|val).
+
+            Return:
+                Tuple: (unarchived img path, unarchived annotation path)
+            """
+
+        archive_img_file = Path(data_path) / f"{split}2017.zip"
+        archive_ann_file = Path(data_path) / "annotations_trainval2017.zip"
+        if archive_img_file.exists() and archive_ann_file.exists():
+            img_file_descriptor = os.open(archive_img_file, os.O_RDONLY)
+            try:
```

Comment:
Violate DRY. It seems line 208-221 is. repeated in 224-238. Can you wrap them in a helper method?

Reply:
Good call. Wrap it into a function. And test it in [kubeflow](https://datasetinsights.endpoints.unity-ai-thea-test.cloud.goog/_/pipeline/#/runs/details/5fd4ad7a-0718-4a23-a034-395800c46569).